native/zowed/server.cpp

```diff
@@ -361,3 +346,46 @@ void RpcServer::send_notification(const RpcNotification &notification)
   string json_string = serialize_json(zjson::to_value(notification).value());
   std::cout << json_string << std::endl;
 }
+
+void RpcServer::print_error(int request_id, int code, const string &message, const string *data)
+{
+  zstd::optional<zjson::Value> error_data;
+  if (data != nullptr)
+  {
+    error_data = zjson::Value(*data);
+  }
+
+  ErrorDetails error{code, message, error_data};
+
+  RpcResponse response;
+  response.jsonrpc = "2.0";
+  response.result = zstd::optional<zjson::Value>();
+  response.error = zstd::optional<ErrorDetails>(error);
+  // Use -1 as sentinel for null ID (per JSON-RPC spec for parse errors)
+  response.id = (request_id == -1) ? zstd::optional<int>() : zstd::optional<int>(request_id);
+
+  print_response(response);
+}
+
+validator::ValidationResult RpcServer::validate_json_with_schema(const string &method, const zjson::Value &data, bool is_request)
+{
+  CommandDispatcher &dispatcher = CommandDispatcher::get_instance();
```

Comment:
Consider updating this to be a constant reference since the logic in this function does not modify state in `dispatcher`